### ISSUE TITLE
Troubles when parsing brace-group + here-document with file descriptor

### ISSUE THREAD

**Issue by Niols** [colis-anr/morbig#23]
When parsing something like:

    { cat;} 9<<EOF
    a
    EOF

Morbig complains:

> line 1, character 11: Syntax error.

It does not happen when there is no brace group or when there is no file descriptor. It seems to me that it should be accepted by the grammar. `dash`, `bash` and `zsh` do accept the script.

I added a test (68a1cd59c938a966b96174054c3128eea6b2bcbd) for this particular case.

**Comment by treinen** [comment]
Yes, I agree that your input should be accepted. The non-terminal io_here may be preceeded by an io_number, according to my reading of the grammar.

Your example looks familiar to me. Didn't we have already a similar bug before?

**Comment by Niols** [comment]
We had a bug with file descriptors, where

    echo 2> foo

was recognized as the words `echo` and `2` and the redirection `> foo`. I don't know if that is what you have in mind. Anyways, I will have a look at the grammar in Menhir. Maybe this bug is obvious there.

**Comment by Niols** [comment]
OK, found it. In fact, it comes from an other bug: in 

    cat 9<<EOF
    a
    EOF

`9` is recognized as a word in the `cmd_suffix` and NOT as an `io_number`. Which explains why we end up with a parse error in our situation, because Morbig does not expect a word there. This is a big strange though, because we already solved that bug for `cat 9>foo`.
